packages/cli/src/controllers/folder.controller.ts

```diff
@@ -86,4 +92,21 @@ export class ProjectController {
 			throw new InternalServerError();
 		}
 	}
+
+	@Get('/')
+	@ProjectScope('folder:list')
+	async listFolders(
+		req: AuthenticatedRequest<{ projectId: string }>,
+		res: Response,
+		@Query payload: ListFolderQueryDto,
+	) {
+		const { projectId } = req.params;
+
+		const [data, count] = await this.folderService.getManyAndCount(
+			projectId,
+			payload as ListQuery.Options,
```

Comment:
removing this type assertion would require some more refactoring in the folder service and repository as in the workflow service and repository